gcd/src/main/java/org/spine3/server/storage/datastore/DatastoreWrapper.java

```diff
@@ -20,220 +20,246 @@
-        }
+    /*package*/ static DatastoreWrapper wrap(Datastore datastore) {
+        return new DatastoreWrapper(datastore);
     }
 
     /**
-     * Commits the given mutation.
+     * Writes new {@link Entity} into the datastore.
      *
-     * @param mutation the mutation to commit.
-     * @throws RuntimeException if {@link Datastore#commit(CommitRequest)} throws a {@link DatastoreException}
+     * @param entity new {@link Entity} to put into datastore
+     * @throws DatastoreException if an {@link Entity} with such {@link Key} already exists.
+     * @see DatastoreWriter#put(FullEntity)
      */
-    @SuppressWarnings("TypeMayBeWeakened") // no, it cannot
-    /* package */ void commit(Mutation.Builder mutation) {
-        commit(mutation.build());
+    @SuppressWarnings("WeakerAccess") // Can be used in future
+    /*package*/ void create(Entity entity) throws DatastoreException {
+        actor.add(entity);
     }
 
     /**
-     * Commits the {@link LookupRequest}.
+     * Modifies an {@link Entity} in the datastore.
      *
-     * @param request the request to commit
-     * @return the {@link LookupResponse} received
-     * @throws RuntimeException if {@link Datastore#lookup(LookupRequest)} throws a {@link DatastoreException}
+     * @param entity the {@link Entity} to update.
+     * @throws DatastoreException if the {@link Entity} with such {@link Key} does not exist.
+     * @see DatastoreWriter#update(Entity...)
      */
-    /* package */ LookupResponse lookup(LookupRequest request) {
-        LookupResponse response = null;
-        try {
-            response = datastore.lookup(request);
-        } catch (DatastoreException e) {
-            propagate(e);
-        }
-        return response;
+    @SuppressWarnings("WeakerAccess") // Can be used in future
+    /*package*/ void update(Entity entity) throws DatastoreException {
+        actor.update(entity);
     }
 
     /**
-     * Runs the given {@link Query}.
+     * Writes an {@link Entity} to the datastore or modifies an existing one.
      *
-     * @param query the query to run.
-     * @return the {@link EntityResult} list received or an empty list
-     * @throws RuntimeException if {@link Datastore#lookup(LookupRequest)} throws a {@link DatastoreException}
+     * @param entity the {@link Entity} to write or update.
+     * @see DatastoreWrapper#create(Entity)
+     * @see DatastoreWrapper#update(Entity)
      */
-    private List<EntityResult> runQuery(Query query) {
-        final RunQueryRequest queryRequest = RunQueryRequest.newBuilder().setQuery(query).build();
-        return runQueryRequest(queryRequest);
+    /*package*/ void createOrUpdate(Entity entity) {
+        actor.put(entity);
     }
 
-    /* package */ Iterator<EntityResult> runQueryForIterator(Query query) {
-        return new PagingDatastoreIterator(query, this, datastoreOptions.getEventIteratorPageSize());
+    /**
+     * Retrieves an {@link Entity} with given key from datastore.
```

Comment:
`the given key`.
Also either `Datastore` or `the datastore`. I'd prefer `Datastore` option.